qpext/README.md

```diff
@@ -0,0 +1,166 @@
+full discussion on this topic). In an attempt to make an easy-to-use solution, the queue-proxy is extended to handle this use case. 
+
+
+see also: [KServe Issue #2645](https://github.com/kserve/kserve/issues/2465), 
+
+## How to use 
+
+Save this file as qpext_image_patch.yaml, update the tag if needed.
+```yaml
+data:
+  queueSidecarImage: kserve/qpext:latest
+```
+
+Run the following command to patch the deployment config in the appropriate knative namespace.
+```shell
+kubectl patch configmaps -n knative-serving config-deployment --patch-file qpext_image_patch.yaml
+```
+
+## Configs
+
+The qpext relies on pod annotations to be set in the InferenceService YAML. If these annotations are set to true, then environment variables will be added to the queue-proxy container. 
+The qpext uses the environment variables to configure which port/path to expose metrics on and which port/path to scrape metrics from in `queue-proxy` and `kserve-container`.
+
+| Annotation                                           | Default | Description |
+|------------------------------------------------------|---------|-------------|
+| serving.kserve.io/enable-metric-aggregation          | false | If true, enables metric aggregation in queue-proxy by setting env vars in the queue proxy container to configure scraping ports. |
+| serving.kserve.io/enable-prometheus-scraping | false | If true, sets the prometheus annotations in the pod. If true and "serving.kserve.io/enable-metric-aggregation" is false, the prometheus port will be set as the default queue-proxy port. If both are true, the prometheus port annotation will be set as the aggregate metric port.  | 
+
+
+| Queue Proxy Env Vars                     | Default  | Description                                                                                                                                                                     |
+|------------------------------------------|----------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| AGGREGATE_PROMETHEUS_METRICS_PORT        | 9088     | The metrics aggregation port in queue-proxy that is added in the qpext.                                                                                                         | 
+| KSERVE_CONTAINER_PROMETHEUS_METRICS_PORT | 8080     | The default metrics port for the `kserve-container`. If present, the default ClusterServingRuntime overrides this value with each runtime's default prometheus port.            |
+| KSERVE_CONTAINER_PROMETHEUS_METRICS_PATH | /metrics | The default metrics path for the `kserve-container`. If present, the default ClusterServingRuntime annotation overrides this value with each runtime's default prometheus path. |   
+
+To implement this feature, configure the InferenceService YAML annotations.
```

Comment:
Do we support the case when user just need to enable the aggregation in the configmap so user do not need to add the annotation to the inference yaml?

Reply:
yes but what is the best way to instruct people to change the config map? right now default is false. 

for docs we need to tell kserve maintainers the best way to implement that change and then tell users how to check defaults (just check configmap). the latter is straight forward, does the former require a patch? not sure if that is the recommended way or not

Reply:
updated - lmk what you think

Reply:
For people who installs kserve with helm chart they can change the helm variables you defined, with yaml installation the kserve admin would need to patch or kustomize the configmap before installing it.

Reply:
Ok I think I added that info here- do you think I should be more explicit ?